src/core/shape_inference/include/interpolate_shape_inference.hpp

```diff
@@ -0,0 +1,228 @@
+    }
+    return static_cast<int64_t>(static_cast<float>(bound) * scale);
+}
+
+template <typename T>
+void infer_using_scales(T& output_shape,
+                        const std::vector<int64_t>& axes,
+                        const std::vector<float>& scales,
+                        const T& padded_input_shape) {
+    size_t i = 0;
+    static constexpr float epsilon = 1.0e-6f;
+    for (auto axis : axes) {
+        const auto& current_dim = padded_input_shape[axis];
+        float multiplier = scales[i] + epsilon;
+        if (current_dim.is_static()) {
+            output_shape[axis] = multiply_bound_and_scale(current_dim.get_length(), multiplier);
+        } else {
+            int64_t new_lower_bound = multiply_bound_and_scale(current_dim.get_min_length(), multiplier);
+            int64_t new_upper_bound = multiply_bound_and_scale(current_dim.get_max_length(), multiplier);
+            set_low_upper_bound(output_shape[axis], new_lower_bound, new_upper_bound);
+        }
+        ++i;
+    }
+}
+
+template <class T>
+void shape_infer(const Interpolate* op,
+                 std::vector<size_t>& pads_begin,
+                 std::vector<size_t>& pads_end,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data) {
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 3 || input_shapes.size() == 4) && output_shapes.size() == 1);
+
+    const auto& input_shape = input_shapes[0];
+    auto& output_shape = output_shapes[0];
+    output_shape = input_shape;
+
+    if (input_shape.rank().is_static()) {
+        const auto input_rank = input_shape.size();
+
+        // Get axes
+        std::vector<int64_t> axes;
+        if (input_shapes.size() == 4 && !(get_data_as_int64<T>(3, op, axes, constant_data))) {
+            for (size_t i = 0; i < input_rank; i++)
+                output_shape[i] = ov::Dimension::dynamic();
+            return;
+        } else if (input_shapes.size() == 3) {
+            axes.resize(input_rank);
+            std::iota(axes.begin(), axes.end(), 0);
+        }
+        NODE_VALIDATION_CHECK(op,
+                              std::all_of(axes.begin(),
+                                          axes.end(),
+                                          [input_rank](size_t axis) {
+                                              return axis < input_rank;
+                                          }),
+                              "Axis value should less than input rank.");
+
+        // Get padded input shape
+        auto padded_input_shape = input_shape;
+        for (int64_t i = 0; i < input_rank; ++i) {
+            if (input_shape[i].is_static()) {
+                auto new_length = pads_begin[i] + pads_end[i] + input_shape[i].get_length();
+                padded_input_shape[i] = new_length;
+            }
+        }
+
+        output_shape = padded_input_shape;
```

Comment:
Looks like we can get rid of `padded_input_shape` and immediately fill `output_shape` there